syntax = "proto3";

package nemo_chem.models.megamolbart.grpc;

// import "google/protobuf/empty.proto";

//python -m pip install grpcio
//python -m pip install grpcio-tools
//python -m grpc_tools.protoc -I./setup/ \
//          --python_out=./generated/ \
//          --grpc_python_out=./generated/ \
//          --experimental_allow_proto3_optional \
//          ./setup/megamolbart.proto

service GenerativeSampler {
  rpc SmilesToEmbedding(InputSpec) returns (OutputSpec) {};
  rpc SmilesToHidden(InputSpec) returns (OutputSpec) {};
  rpc HiddenToSmis(InputSpec) returns (OutputSpec) {};
}


message InputSpec {
  repeated string smis = 1;
  repeated float hidden_states = 2;
  repeated int32 dim = 3;
  repeated bool masks = 4;
}


message OutputSpec {
  repeated string smis = 1;
  repeated float hidden_states = 2;
  repeated float embeddings = 3;
  repeated int32 dim = 4;
  repeated bool masks = 5;
}
